### PR TITLE
Transpose nullspace

### PR DESCRIPTION
**Pull request by wence-**
Needs https://bitbucket.org/petsc/petsc4py/pull-requests/72/expose-matxxxtransposenullspace/diff#comment-18413593 first.

### CODE REVIEW

**Comment by wence-** [comment]
An eyeball of the testing would be good.  Seems to work for Francis Poulin's QG demo as well.

**Comment by colinjcotter** [comment]
I'd like to talk to about whether we need this for the Monge-Ampere problem.

cheers
--cjc

On 12 May 2016 at 19:00, Lawrence Mitchell <notifications@github.com> wrote:

> An eyeball of the testing would be good. Seems to work for Francis
> Poulin's QG demo as well.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/firedrakeproject/firedrake/pull/778#issuecomment-218836894>
>



-- 
http://www.imperial.ac.uk/people/colin.cotter

www.cambridge.org/9781107663916

**Comment by wence-** [comment]
Sure.

File: docs/source/solving-interface.rst [hunk 1]
```diff
@@ -715,6 +715,14 @@ the solution to be :math:`u(x, y) = y - 0.5`.
...
 
+For this to work, the provided right hand side must be orthogonal to
+the transpose nullspace of the operator as well.  In many cases, we
+can arrange for this to occur by careful choice of initial
+conditions.  Sometimes this is not possible.  In this case, you can
+ask Firedrake to remove the component of the right hand side that is
+in the transpose nullspace by providing a
+:class:`~firedrake.nullspace.VectorSpaceBasis` with the
+``transpose_nullspace`` keyword argument to :func:`~.solve`.
 Singular operators in mixed spaces
```
Comment (miklos1) [inline docs/source/solving-interface.rst]: Empty line before title?

**Pull request merged**